addons/resource/models/resource.py

```diff
@@ -958,7 +958,7 @@ def _adjust_to_calendar(self, start, end):
                 # Make sure to only search end after start
                 search_range = (
                     start,
-                    end + relativedelta(days=1, hour=0, minute=0, second=0),
+                    end + relativedelta(hour=23, minute=59, second=59),
```

Comment:
I didn't try to play with this part of the code but what if the issue was a timezone issue ?

I explain my pov : `end` is not aware of the timezone `tz` of the user (it's in UTC), if we had one day and force the values of hour, minute and second to the start of the day, we may have something unexpected : 
- let's say the user is in belgium (Summer DST so UTC+2) :
- start is : 22:00:00 UTC on Day 1
- end is : 23:59:59 UTC on Day 1
Those dates have the same date, with timezone Europe/Brussels (see condition line 957) : Day 2

If I make : `end + relativedelta(days=1, hour=0, minute=0, second=0)`, I'll obtain 00:00:00 UTC on Day 2 because the timezone of end is UTC and not Europe/Brussels, and I think the goal is to obtain something like : 22:00:00 UTC on Day 2, no ?

Do you understand my charabia ? Does this make sense regarding what you analyzed ?

Reply:
Thus I would suggest something like that :

```suggestion
                    end.astimezone(tz) + relativedelta(hour=23, minute=59, second=59),
```

Or, what I think is better and clearer :
```suggestion
                    start.astimezone(tz) + relativedelta(hour=23, minute=59, second=59),
```